src/state-manager/index.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+import {
+  updateBrowserHistoryStack,
+  getActiveStepFromUrl,
+} from './url-state-manager';
+
+const StateManager = (mode: string) => {
+  const supportedModes = ['url'];
+  let selectedMode = '';
+
+  if (supportedModes.includes(mode)) {
+    selectedMode = mode;
+  } else {
+    throw new Error(
+      `${mode} is not supported as a mode, use one of ${supportedModes.join(
+        ', ',
+      )}`,
+    );
+  }
+
+  return {
+    updateHistory(currentStepName: string, nextStepName: string) {
+      switch (selectedMode) {
+        case 'url': {
+          updateBrowserHistoryStack(currentStepName, nextStepName);
+          break;
+        }
+        default:
```

Comment:
You will never reach this code, so you don't need the switch. Perhaps you added it to more easily add a new mode?

Reply:
Yup